pandas/tests/extension/base/getitem.py

```diff
@@ -120,3 +121,11 @@ def test_take_sequence(self, data):
         assert result.iloc[0] == data[0]
         assert result.iloc[1] == data[1]
         assert result.iloc[2] == data[3]
+
+    def test_take_empty_missing(self, data, na_value, na_cmp):
+        empty = data[:0]
+        result = empty.take([-1])
+        na_cmp(result[0], na_value)
+
+        with tm.assert_raises_regex(IndexError, "cannot do a non-empty take"):
+            empty.take([0, 1])
```

Comment:
@TomAugspurger I didn't find any existing tests for `take`, so not sure the indexing tests is the best place (maybe rather the `BaseMethodsTests`). 
And, I could also add tests for the actual use cases where you get this (eg `reindex` on an empty series)

Reply:
This seems like the right place for testing a behavior specific to take.